### PR TITLE
A-AddCustomers

### LINKED ISSUE #48: Implement Customer class
Add support for initialising a customer object containing username , age and contact number. Getter and Setter Methods as well.

### LINKED ISSUE #49: Implement parser class to parse "add-user" command
Parse add-user input according to parameters set in UG

### LINKED ISSUE #50: Add simple exceptions and javadoc to Customer and Parser class
Header comments and necessary exceptions

### LINKED ISSUE #51: Write JUnit test for the Customer and Parser class
Junit test for methods related to add-user command in Parser and Customer Class

### PR DESCRIPTION
**Pull request by rexkoh425**
Implemented the Customer, CustomerList, and Parser Class. Fixes #48  Fixes #49 
Added Javadoc and exceptions to the methods. Fixes #50 
Implemented Junit test for add-user commands. Fixes #51 

Fixes #52

### CODE REVIEW

File: src/main/java/customer/Customer.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package customer;
+
+/**
+ * Represents customers of CliRental
+ */
+public class Customer {
+
+    private String username;
+    private int age;
+    private int contactNumber;
```
Comment (CT9ARyan) [inline src/main/java/customer/Customer.java:10]: Perhaps we can store contactNumber as a String??

File: src/main/java/customer/Customer.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    public Customer(String username , int age , int contactNumber){
```
Comment (CT9ARyan) [inline src/main/java/customer/Customer.java:12]: Remove the trailing spaces after "username" and "age".

File: src/main/java/parser/Parser.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+import customer.Customer;
+import exceptions.CustomerException;
+import java.util.Scanner;
+
+public class Parser {
+
+    public static Scanner scanner = new Scanner(System.in);
+    public static String userInput;
+    private static final String addCustomerCommand = "add-user";
```
Comment (CT9ARyan) [inline src/main/java/parser/Parser.java:11]: Constants should be in SCREAMING_SNAKE_CASE. (i.e. ADD_CUSTOMER_COMMAND)

File: src/main/java/parser/Parser.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+     * Creates new customer object based on user input.
+     *
+     * @throws CustomerException if input is not compliant with format.
+     * @throws NumberFormatException if the age and contact content are not integer string.
+     */
+    public static Customer parseIntoCustomer() throws CustomerException , NumberFormatException {
+        userInput = userInput.substring(addCustomerCommand.length()).trim();
+        String[] parameters = { "/u" , "/a" , "/c"};
+        String[] parameterContents;
+        if(isValidSequence(parameters)){
```
Comment (CT9ARyan) [inline src/main/java/parser/Parser.java:36]: Maybe you can follow the "make the happy path prominent" coding standard here?? (i.e. Use the guard clause "if( !isValidSequence(parameters) ), then throw the exception here in the if block.

File: src/main/java/seedu/duke/Duke.java [hunk 5]
```diff
@@ -1,21 +1,55 @@
 package seedu.duke;
 
-import java.util.Scanner;
+import customer.Customer;
+import customer.CustomerList;
+import exceptions.CustomerException;
+import parser.Parser;
 
 public class Duke {
```
Comment (CT9ARyan) [inline src/main/java/seedu/duke/Duke.java:8]: Change "Duke" to "CLIRental"

**Review by CT9ARyan — COMMENTED** [review]
LGTM 👍

File: src/main/java/customer/Customer.java (hunk 1 above)
Comment (AaronZZ10) [reply src/main/java/customer/Customer.java:10]: String would be better, or at least Long

**Pull request merged**